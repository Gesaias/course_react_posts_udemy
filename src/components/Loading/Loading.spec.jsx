import { render, screen } from '@testing-library/react';
import { Loading } from '.';

describe('<Loading />', () => {
  it('should render Loading correctly non-centered', () => {
    render(<Loading center={false} />);

    expect.assertions(1);

    expect(screen.getByText(/Loading/i)).toBeInTheDocument();
  });

  it('should render Loading correctly centered', () => {
    render(<Loading center={true} />);

    expect.assertions(1);

    expect(screen.getByText(/Loading/i)).toHaveAttribute('class', 'center');
  });

  // Snapshot

  it('should match snapshot', () => {
    const container = render(<Loading center={true} />).container;
    const { firstChild } = container;

    expect(firstChild).toMatchSnapshot();
  });
});
