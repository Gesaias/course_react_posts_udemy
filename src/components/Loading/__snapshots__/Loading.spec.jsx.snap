// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Loading /> should match snapshot 1`] = `
<p
  class="center"
>
  Loading...
</p>
`;
